### PR TITLE
[AIRFLOW-6413] Add config file for Mergeable Github integration

### PR DESCRIPTION
**Pull request by kaxil**
Add config for  https://github.com/apps/mergeable

Jira: https://issues.apache.org/jira/browse/AIRFLOW-6413

- [x] Description above provides context of the change
- [x] Commit message contains [\[AIRFLOW-XXXX\]](https://issues.apache.org/jira/browse/AIRFLOW-XXXX) or `[AIRFLOW-XXXX]` for document-only changes
- [x] Unit tests coverage for changes (not needed for documentation changes)
- [x] Commits follow "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)"
- [x] Relevant documentation is updated including usage instructions.
- [x] I will engage committers as explained in [Contribution Workflow Example](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#contribution-workflow-example).
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).
Read the [Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines) for more information.

### CODE REVIEW

**Review by potiuk — APPROVED** [review]

File: .github/mergeable.yml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    validate:
+      - do: title
+        # Do not merge when it is marked work in progress (WIP)
+        must_exclude:
+          regex: ^\[WIP\]
+          message: This is work in progress. Do not merge yet.
+        begins_with:
+          match: '[AIRFLOW-'
+        must_include:
+          regex: ^(\[AIRFLOW-XXXX\]|\[AIRFLOW\-\d{1,4}\])
```
Comment (ashb) [inline .github/mergeable.yml:30]: This should be `AIRFLOW-XXX` not `XXXX` :)

Comment (kaxil) [reply .github/mergeable.yml:30]: :D This was intentional as there was a recent change in Github PR template that asked users to add 4 X's `[AIRFLOW-XXXX]` for docs.

Comment (potiuk) [reply .github/mergeable.yml:30]: That reminds you that if you have XXX in the release scripts to distinguish doc-only updates, it should be updated to support also XXXX (maybe - depends if you have in search pattern "]" ;)

Comment (potiuk) [reply .github/mergeable.yml:30]: BTW. I really love that people will have to manually click the checkboxes now (though initially people won't understand why mergable complains). Happened already with @michalslowikowski00 :D. But I think this is good if they ask.

**Pull request merged**